include/Piece.h

```diff
@@ -0,0 +1,69 @@
+#pragma once
+#include "Color.h"
+#include <vector>
+#include "Location.h"
+
+/// <summary>
+/// Abstract class of chess piece
+/// </summary>
+class Piece
```

Comment:
Inheritance is not good enough - all tools inherit from Piece. Intermediate classes must be created for walking diagonally and walking straight, thus giving a finer inheritance for different tools.